airflow/models.py

```diff
@@ -3858,7 +3856,10 @@ def find(dag_id=None, run_id=None, execution_date=None,
         if run_id:
             qry = qry.filter(DR.run_id == run_id)
         if execution_date:
-            qry = qry.filter(DR.execution_date == execution_date)
+            if isinstance(execution_date, list):
```

Comment:
I meant that if execution_dates is always a list (e.g. [2016-04-10]) then you don't need this if statement or the else